modules/calib3d/include/opencv2/calib3d.hpp

```diff
@@ -3270,14 +3270,1282 @@ optimization. It stays at the center or at a different location specified when C
+
+/** @brief isPlanarTag Checks if the object points are in the correct format of a planar tag
+    @param _opoints
+    @return true if correct, false otherwise
+    */
+    CV_WRAP bool isPlanarTag(InputArray _opoints) const;
+
+
+protected:
+
+/** @brief Makes a 3x3 identity intrinsic matrix
+
+    @returns Identity intrinsic matrix (3x3 single channel double)
+    */
+    CV_WRAP cv::Mat makeIdentityIntrinsic() const;
+
+/** @brief getNumberOfPoints Gets number of points in object matrix. Must be 1xN 3-channel or Nx1 3-channel matrix (double) where n is the number of points.
+
+    @param opoints Set of Object points. This must be a 1xN 3-channel or Nx1 3-channel matrix (double) where n is the number of points.
+    opoints are defined in object coordinates.
+    @return
+    */
+    CV_WRAP size_t getNumberOfPoints(InputArray opoints) const;
+
+
+/** @brief Checks that argument types are correct. Specifically opoints must be a 1xN 3-channel or Nx1 3-channel matrix (double) where n is the number of points.
+    opoints are defined in object coordinates. ipoints must be a 1xN 2-channel or Nx1 2-channel matrix (double) where n is the number of points.
+
+    @returns true if correct, otherwise false
+    */
+    CV_WRAP void checkArgTypes(InputArray opoints, InputArray ipoints) const;
+
+
+/** @brief Checks if the problem geometry is known to cause an artificial degeneracy.
+    An artificial degeneracy is one where the problem is theoretically solvable,
+    but the algorithm design prevents it from solving the problem. To know more about
+    artifical degeneracies, see e.g. @cite Collins14
+
+    @returns true if there not an artificial degeneracy, false otherwise
+    */
+    CV_WRAP virtual bool noArtificialDegeneracy(InputArray opoints,InputArray ipoints) const;
+
+/** @brief Solves PnP for a given set of inputs
+
+    @param opoints Set of Object points. This must be a 1xN 3-channel or Nx1 3-channel matrix (double) where n is the number of points.
+    opoints are defined in object coordinates.
+    @param ipoints Set of Image points. This must be a 1xN 2-channel or Nx1 2-channel matrix (double) where n is the number of points.
+    ipoints are defined in normalized pixel coordinates. That is, they are defined by using a canonical intrinsic calibration matrix of K = eye(3).
+    You can convert a set of points defined in pixels to normalized pixel coordinates using @ref undistortPoints. This is convenient because it
+    eliminates the need for supplying the camera's intrinsics (calibration matrix and distortion parameters)
+    @param rvecs Vector of output rotation vectors (double) (see @ref Rodrigues ) that, together with tvecs, brings points from
+    the model coordinate system to the camera coordinate system.
+    @param tvecs Vector of output translation vectors (double)
+     */
+    CV_WRAP virtual void solve(InputArray opoints, InputArray ipoints, CV_OUT std::vector<Mat> & rvecs, CV_OUT std::vector<Mat> & tvecs) const = 0;
```

Comment:
Why is this method protected? (I believe `solveProblem()` should be virtual instead)